### ISSUE TITLE
Use log scale in gauges?

### ISSUE THREAD

**Issue by waldyrious** [edsu/wikipulse#13]
The visualization could be a lot more interesting if the gauges used a logarithmic scale. Currently most of the pointers linger near zero, and a couple of more active projects hang consistently around middle-ground values, with occasional spikes that send them over the limit. Increasing the limit would create some room for these spikes, but would glue the less active projects even more to the zeroth position; decreasing the limit would make the fluctiations of the small projects more visible, at the expense making off-the-scale events unacceptably frequent for the larger projects.

A log scale would fix both issues neatly. Unfortunately, Google Charts doesn't support a log scale for the gauge chart type, but I found a web chart library, called [HighCharts](http://www.highcharts.com/), [that does](http://api.highcharts.com/highcharts#yAxis.type). It's free for non-commercial use, so we should be fine using it. Here's how it would look: http://jsfiddle.net/8LB8V/. What do you say, @edsu?

**Comment by edsu** [comment]
I think it might actually be useful to switch from using gauges to something like http://bost.ocks.org/mike/path/ which would provide a view into the near past. A log scale would work well eith these I think?

**Comment by waldyrious** [comment]
Oh, that'd be lovely! I thought the use of gauges was deliberate because preserving a window into the past would require too much memory, so yeah, I'm all for it :)

That said, a [polygraph](http://ak.picdn.net/shutterstock/videos/1094638/preview/stock-footage-seismograph-recorded-volcanic-activity.jpg)-[like](http://2.bp.blogspot.com/-mwLekMe20DQ/UQp8eATjxxI/AAAAAAAAC1M/k-ZrD8-vs0o/s1600/4.jpg) visualization would surely be nice but I fear it would be harder to fit so many streams into a single page, as well as to compare different projects at a glance (also it'd be less visually interesting than the gauge display). How about having a single line chart below (or above) the gauges, showing the global activity by default, and changing to individual projects when we hover their corresponding gauges?

**Comment by waldyrious** [comment]
Update: I finally went ahead and implemented this, along with a revamped infrastructure based on the new [RCFeed](https://www.mediawiki.org/wiki/Manual:RCFeed) service at stream.wikimedia.org. So now the whole thing is server-less, yay!

Check it out! --> http://waldyrious.net/wikispeed/

**Comment by edsu** [comment]
It looks great!

**Comment by edsu** [comment]
Is the code you are running worth merging in here? Or is it all new?

**Comment by waldyrious** [comment]
@edsu it's all new, that's why I created a separate project and gave it a new name as I felt I couldn't simply "appropriate" the name Wikipulse and provide the same service elsewhere. Functionally, however, it's exactly the same, so if you would like to replace the current wikipulse codebase with the new one, and maybe call it "Wikipulse 2.0" or something like that, I could certainly see the benefit of not having two dashboards doing identical things.

One of the greatest advantages of the setup I went for, I think, is a completely server-less infrastructure: the whole thing even runs if you save the page locally and open the file in the browser! That makes the whole architecture simpler, and also makes it possible to host the page directly on github :)

That said, I'll defer to your judgment on how to proceed.

**Comment by edsu** [comment]
Yes I love that the server side piece is now part of wikipedias infrastructure and not a one off project. Is your code currently on GitHub?

**Comment by waldyrious** [comment]
Yes, of course :) it is hosted in Github Pages, even -- the only reason the link lives at waldyrious.net is because my personal site also hosted on Github, and its CNAME domain setting makes all my repo pages also redirect there. I did add the link to the repo to the intro paragraph, but perhaps a github icon would make it more evident? Anyway, here's the repo: https://github.com/waldyrious/wikispeed

ps - I did licence it under the permissive ISC license (functionally equivalent to the MIT, but shorter), but I can release the code in the public domain if you prefer it.
